### ISSUE TITLE
Syncfusion PDF Viewer - Some PDF file are failing to load and showing a grey screen

### ISSUE THREAD

**Issue by aranbit** [syncfusion/flutter-widgets#1966]
### Bug description

I have a Flutter Web app that allows uploading and displaying pdf files using Syncfusion PDF Viewer.

This has been working successfully until recently when files from some source were failing to load consistently.

**The error log is:**
Error
There was an error opening this document.

[G240672711145675000.f94360ipDMia_TIF.pdf](https://github.com/user-attachments/files/16122927/G240672711145675000.f94360ipDMia_TIF.pdf)

### Steps to reproduce

I have attached a file that should allow you to reproduce this issue.

The following is the code I use to show the PDF:
```
SfPdfViewer.network(
    url,
    onDocumentLoadFailed: (e) {
        debugPrint('Error loading PDF: ${e.error}');
    },
)
```

In my index.html I have these 2 scripts:
```
<script src="//cdnjs.cloudflare.com/ajax/libs/pdf.js/2.11.338/pdf.min.js"></script>
<script type="text/javascript">
      pdfjsLib.GlobalWorkerOptions.workerSrc =
        "//cdnjs.cloudflare.com/ajax/libs/pdf.js/2.11.338/pdf.worker.min.js";
</script>
```

Like I've said this has been working and still is for every file I've tested, except for the kind I have attached in this issue.

### Code sample

<details open><summary>Code sample</summary>

```dart
[Add your code here]
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Stack Traces

<details open><summary>Stack Traces</summary>

```dart
[Add the Stack Traces here]
```

</details>

### On which target platforms have you observed this bug?

Web

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[Add your output here]
```

</details>

**Comment by FetFrumos** [comment]
I have same issue

**Comment by immankumarsync** [comment]
Hi @aranbit,
We are able to replicate the issue, "Exception occurs when loading a specific document". Currently, we are validating this issue, and we will update further details once the validation is completed.

**Comment by immankumarsync** [comment]
The fix for the reported issue "Type cast issue occurs when retrieving the annotations in a specific PDF document" has been included in our weekly release v26.1.42. 

Kindly use the following link to refer to the latest release version,

[syncfusion_flutter_pdf package - All Versions (pub.dev)](https://pub.dev/packages/syncfusion_flutter_pdf/versions) 

Root cause: To retrieve annotations from a PDF page, we parse the annotation entry from the page dictionary. However, we encountered a type cast issue in a scenario where the PDF document has a null value for annotations in its page dictionary.

**Comment by aranbit** [comment]
I have checked this, and it seems to be now working for me!!!
Thank you very much for fixing this issue this fast 🙏
